signac/core/validators.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) 2017 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"""Validators for SyncedCollection API."""
+import warnings
+from collections.abc import Mapping
+from collections.abc import Sequence
+
+from ..errors import InvalidKeyError
+from ..errors import KeyTypeError
+from .synced_collection import NUMPY
+
+if NUMPY:
+    import numpy
+
+
+class NoDotInKey:
```

Comment:
I don't think either of the validators in this file need to be classes, I would implement them as functions.